Bottom: 675b0d3f0525e2f592b568ab613b59e3555298de
Top:    614520780a8b216704eef3e883b67cee0d45c7fc
Author: Johannes Rudolph <johannes_rudolph@gmx.de>
Date:   2009-08-16 12:25:34 +0200

ojf: throw (not-yet) well-known exception on parse errors


---

diff --git a/objectformatter/src/main/scala/net/virtualvoid/string/ESParser.scala b/objectformatter/src/main/scala/net/virtualvoid/string/ESParser.scala
index a4f4da1..f396b3c 100644
--- a/objectformatter/src/main/scala/net/virtualvoid/string/ESParser.scala
+++ b/objectformatter/src/main/scala/net/virtualvoid/string/ESParser.scala
@@ -100,6 +100,8 @@ object AST{
   }
 }
 
+class ParseException(msg:String) extends Exception(msg)
+
 object EnhancedStringFormatParser extends RegexParsers{
   import AST._
   
@@ -159,6 +161,6 @@ object EnhancedStringFormatParser extends RegexParsers{
   def parse(input:String):FormatElementList = 
     phrase(tokens)(new scala.util.parsing.input.CharArrayReader(input.toCharArray)) match {
       case Success(res,_) => res
-      case x:NoSuccess => error(x.msg)
+      case x:NoSuccess => throw new ParseException(x.msg)
     }
 }
\ No newline at end of file
